File: messenger_app/lib/features/chat/data/dto/message_request.dart
class MessageRequest {
  final String? content;
  final String? fileUrl;
  final String type;
  final int? replyToId;

  MessageRequest({
    this.content,
    this.fileUrl,
    this.type = 'TEXT',
    this.replyToId,
  });

  Map<String, dynamic> toJson() => {
        if (content != null) 'content': content,
        if (fileUrl != null) 'fileUrl': fileUrl,
        'type': type,
        if (replyToId != null) 'replyToId': replyToId,
      };
}

File: msgr-chat/src/main/java/ru/vsu/cs/msgr_chat/dto/MessageRequest.java
package ru.vsu.cs.msgr_chat.dto;

import lombok.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MessageRequest {
    private String content;
    private Long chatId;
    private String fileUrl;
    private String type;        // ← ДОБАВЛЕНО: TEXT, IMAGE, FILE
    private Long replyToId;
}